docs/source/tutorial/multi_node_multi_gpu_tutorial.rst

```diff
@@ -0,0 +1,252 @@
+
+    import argparse
+    import time
+    import warnings
+
+    import torch
+    import torch.distributed as dist
+    from torch_geometric.datasets import FakeDataset
+
+    from torch_geometric.nn.models import GCN
+
+    warnings.filterwarnings("ignore")
+
+
+    _LOCAL_PROCESS_GROUP = None
+
+
+    def create_local_process_group(num_workers_per_node):
+        ...
+
+
+    def get_local_process_group():
+        ...
+
+
+    def run(device, data, world_size, model, epochs, batch_size, fan_out,
+                  split_idx, num_classes):
+        ...
+
+
+    if __name__ == '__main__':
+        parser = argparse.ArgumentParser()
+        parser.add_argument('--hidden_channels', type=int, default=64)
+        parser.add_argument('--lr', type=float, default=0.01)
+        parser.add_argument('--epochs', type=int, default=3)
+        parser.add_argument('--batch_size', type=int, default=128)
+        parser.add_argument('--fan_out', type=int, default=50)
+        parser.add_argument(
+            "--ngpu_per_node",
+            type=int,
+            default="1",
+            help="number of GPU(s) for each node for multi-gpu training,",
+        )
+        args = parser.parse_args()
+        # setup multi node
+        torch.distributed.init_process_group("nccl")
+        nprocs = dist.get_world_size()
+        create_local_process_group(args.ngpu_per_node)
+        local_group = get_local_process_group()
+        device_id = dist.get_rank(
+            group=local_group) if dist.is_initialized() else 0
+        torch.cuda.set_device(device_id)
+        device = torch.device(device_id)
+
+        dataset = FakeDataset(avg_num_nodes=100000)
+        data = dataset.data
+        num_nodes = data.num_nodes
+        rand_id = torch.randperm(num_nodes)
+
+        # 60/20/20 split
+        split_idx = {
+            'train':rand_id[:int(.6 * num_nodes)],
+            'valid':rand_id[int(.6 * num_nodes):int(.8 * num_nodes)],
+            'test':rand_id[:int(.8 * num_nodes):],
+        }
+
+        model = GCN(dataset.num_features, args.hidden_channels, 2,
+                    dataset.num_classes)
+        run(device, data, nprocs, model, args.epochs, args.batch_size,
+                  args.fan_out, split_idx, dataset.num_classes)
+
+
+Similarly to the warm up example, we define a :meth:`run` function. However, in this case we are using torch distributed with NVIDIA NCCL backend, instead of relying on :class:`~torch.multiprocessing`. Because we are running on multiple nodes, we want to set up a local process group for each node, and use :obj:`args.ngpu_per_node` GPUs per node. We then select the the CUDA device that will be used by each process within each process group. The next steps are fairly basic :pyg:`PyG` and :pytorch:`PyTorch` usage. We load our (synthetic) dataset and then set up our 60/20/20 train/val/test split. Next, we define our :class:`~torch_geometric.nn.models.GCN` model and finally call our :meth:`run` function.
```

Comment:
```suggestion
Similarly to the single-node multi-gpu example, we define a :meth:`run` function. However, in this case we are using torch distributed with NVIDIA NCCL backend, instead of relying on :class:`~torch.multiprocessing`. Because we are running on multiple nodes, we want to set up a local process group for each node, and use :obj:`args.ngpu_per_node` GPUs per node. We then select the the CUDA device that will be used by each process within each process group. The next steps are fairly basic :pyg:`PyG` and :pytorch:`PyTorch` usage. We load our (synthetic) dataset and then set up our 60/20/20 train/val/test split. Next, we define our :class:`~torch_geometric.nn.models.GCN` model and finally call our :meth:`run` function.
```

Also, I think we should breifly explain or add a link to some documentation about process group as I believe the target readers are someone who doesn't have much experience in multi-node multi-gpu training.